File: GFG/Nth node from end of linked list.java
class Node {
    int data;
    Node next;

    Node(int d) {
        data = d;
        next = null;
    }
}

class GfG {
    // Function to find the data of nth node from the end of a linked list.
    int getNthFromLast(Node head, int n) {
        Node cur = head;

        while (n > 0 && cur != null) {
            cur = cur.next;
            n -= 1;
        }

        if (n > 0) {
            return -1;
        }

        Node ans = head;
        while (cur != null) {
            cur = cur.next;
            ans = ans.next;
        }

        return ans.data;
    }
}

File: GFG/Nth node from end of linked list.py
def getNthFromLast(head, n):
    cur = head
    ans = head
    while n > 0 and cur:
        cur = cur.next
        n -= 1

    if n > 0:
        return -1

    while cur:
        ans = ans.next
        cur = cur.next

    return ans.data
